lib/common/events.ts

```diff
@@ -54,12 +54,12 @@ export function patchEventTarget(
     // invoke static task.invoke
     task.invoke(task, target, [event]);
   };
-
-  // global shared zoneAwareCallback to handle all event callback with capture = false
-  const globalZoneAwareCallback = function(event: Event) {
+  
+  // global shared zoneAwareCallback to handle all event callback with possible capture
+  const globalZoneAwareCallbackBase = function(event: Event, capture: string) {
```

Comment:
@kgentes , yes, the original version of `zone.js` has a much better abstraction and maintenance, no duplicate code, everything have a much better structure.

but the performance become the main issue, so in this case, performance is the essential priority. I think we have to do some trade off, for this `events.ts`, I think we should just focus on performance.